internal/flink/command_statement_list.go

```diff
@@ -74,12 +76,12 @@ func (c *command) statementList(cmd *cobra.Command, args []string) error {
 	}
 
 	list := output.NewList(cmd)
-	for _, statement := range statements {
-		// filter out statements with a non-matching status
-		if status != "" && statement.Status.GetPhase() != status {
-			continue
-		}
 
+	statementsWithMatchingStatus := lo.Filter(statements, func(stmt v1beta1.SqlV1beta1Statement, _ int) bool {
+		return status == "" || stmt.Status.GetPhase() == status
```

Comment:
The check for `status == ""` should be moved outside of this function. No need to check it on every iteration.